eclipse/src/saros/editor/EditorManager.java

```diff
@@ -1079,12 +1071,11 @@ private void replaceText(SPath path, int offset, String replacedText, String tex
           ITextEditor textEditor = (ITextEditor) editorPart;
           IAnnotationModel model =
               textEditor.getDocumentProvider().getAnnotationModel(textEditor.getEditorInput());
+
           contributionAnnotationManager.insertAnnotation(model, offset, text.length(), source);
         }
       }
 
-      IAnnotationModel model = provider.getAnnotationModel(input);
```

Comment:
Do you know why there was a separate handling beforehand? I.e. why is this no longer necessary?

Reply:
Nobody really knows that. As I already said the logic is buggy, annotations are inserted twice (the contribution manager silently ignores it).

Guess it makes more sense either to remove this line or count the editors in the loop above and only execute this code here if you annotated 0 editors.

Reply:
And I guess nobody know if annotation removal from the history is correctly working when the annotation is used in multiple editors (e.g splitview)